XMLHttpRequest/send-redirect-to-cors.htm

```diff
@@ -4,65 +4,64 @@
     <title>XMLHttpRequest: send() - Redirect to CORS-enabled resource</title>
     <script src="/resources/testharness.js"></script>
     <script src="/resources/testharnessreport.js"></script>
-    <link rel="help" href="https://xhr.spec.whatwg.org/#infrastructure-for-the-send()-method" data-tested-assertations="following::dl[1]/dt[2] following::dl[1]/dd[2]/ol/li[1] following::dl[1]/dd[2]/ol/li[3]" />
   </head>
   <body>
     <div id="log"></div>
     <script>
       function extractBody(body) {
         if (body === null) {
           return { body: "", type: "NO" };
-        }
-
-        if (typeof body == "string") {
+        } else if (typeof body == "string") {
           return { body: body, type: "text/plain;charset=UTF-8" };
-        }
-
-        if (body instanceof Uint8Array) {
+        } else if (body instanceof Uint8Array) {
           var arr = Array.prototype.slice.call(body);
           return { body: String.fromCharCode.apply(null, arr), type: "NO" }
+        } else {
+          return { body: "EXTRACT NOT IMPLEMENTED", type: "EXTRACT NOT IMPLEMENTED" }
         }
-
-        return { body: "EXTRACT NOT IMPLEMENTED",
-                 type: "EXTRACT NOT IMPLEMENTED" }
       }
 
-      function redirect(code, name = code, method = "GET", body = null, setExplicitType = true) {
-        var test = async_test(document.title + " (" + name + ")")
-        test.step(function() {
+      function redirect(code, name = code, method = "GET", body = null, setExplicitType = false) {
+        async_test(t => {
           var client = new XMLHttpRequest()
-          client.onreadystatechange = function() {
-            test.step(function() {
-              if (client.readyState == 4) {
+          client.onreadystatechange = t.step_func(() => {
+            if (client.readyState == 4) {
+              if(!setExplicitType) {
                 assert_equals(client.status, 200);
-                assert_equals(client.getResponseHeader("x-request-method"),
-                              method);
+                assert_equals(client.getResponseHeader("x-request-method"), method);
                 var { body: expectedBody, type: expectedType } = extractBody(body);
-                if (setExplicitType) {
-                  expectedType = "application/x-pony";
-                }
-                assert_equals(client.getResponseHeader("x-request-content-type"),
-                              expectedType);
-                assert_equals(client.getResponseHeader("x-request-data"),
-                              expectedBody);
-                test.done();
+                assert_equals(client.getResponseHeader("x-request-content-type"), expectedType);
+                assert_equals(client.getResponseHeader("x-request-data"), expectedBody);
+              } else {
+                // Content-Type is not safelisted by corsenabled.py -> network error
```

Comment:
We have an example that makes an explicit setRequestHeader with text/plain as the value?